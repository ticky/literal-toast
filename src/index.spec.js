/* global describe, it, expect */
import literalToast from './index';

describe('🍞 Literal Toast', () => {
  it('throws if not given the type of arguments it expects', () => {
    expect(() => literalToast()).toThrow();
    expect(() => literalToast('a')).toThrow();
    expect(() => literalToast(3)).toThrow();
    expect(() => literalToast(NaN)).toThrow();
  });

  it('bakes a string literal correctly', () => {
    expect(
      literalToast`This string literal contains some ${'weird substitutions'},
        like ${Infinity}?! and ${NaN}. It's also got some newlines,
        numbers (like ${9}) and is generally kind of unweildy!`
    ).toMatchSnapshot();
  });
});
