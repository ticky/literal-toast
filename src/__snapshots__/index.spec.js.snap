exports[`🍞 Literal Toast bakes a string literal correctly 1`] = `
"This string literal contains some weird substitutions,
        like Infinity?! and NaN. It\'s also got some newlines,
        numbers (like 9) and is generally kind of unweildy!"
`;
